avae/vis.py

```diff
@@ -1145,6 +1145,122 @@ def recon_plot(img, rec, label, data_dim, mode="trn", epoch=0, writer=None):
         logging.info("\n")
 
 
+def latent_4enc_interpolate_plot(
+    x, xs, ys, vae, device, data_dim, plots_config, poses=None
+):
+    """Visualise the interpolation of latent space between 4 randomly selected encodings.
+    The number of plots and the number of interpolation steps is modifyable.
+
+    Parameters
+    ----------
+    x: torch.Tensor
+        A sample batch. we extract 4 random images from this
+    xs: list
+        the list of all latent vectors
+    ys: list
+        the list of all labels for each latent vector in xs
+    vae: torch.nn.Module
+        Affinity vae model.
+    device: torch.device
+        Device to run the model on.
+    data_dim: int
+        the size of spatial dimensions of the input data (2:2D, 3:3D)
+    plots_config: List
+        A list containing the number of plots to be generated and the number of interpolation steps.
+    poses: list
+        List of pose vectors.
+    """
+    logging.info(
+        "################################################################",
+    )
+    logging.info(
+        "Visualising Latent Interpolation between 4 randomly selected encodings ...\n"
+    )
+    padding = 0
+    enc = []
+    decoded_images = []
+    dsize = x[0].shape[-data_dim:]
```

Comment:
Batch is a redundant argument for this function since this is its only use, and `dsize` can be equally inferred from decoded images instead.

Reply:
Agree